### PR TITLE
[CONFORMANCE][IE TESTS] Reduce SLT execution time

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - after:
 ```
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from conformance_other/ReadIRTest
[ RUN      ] conformance_other/ReadIRTest.ReadIR/PRC=_IR_name=Add_4560d07b463570c1fe7087406160cf403db2a3f68a36731018d8c4b9cc871944.xml_TargetDevice=TEMPLATE_Config=()

MEM_USAGE=70356KB
Access PluginCache ov core. OV Core use count: 0
Created ov core.
 rel_threshold: 1.79769e+308
abs_threshold: 0.30426
[       OK ] conformance_other/ReadIRTest.ReadIR/PRC=_IR_name=Add_4560d07b463570c1fe7087406160cf403db2a3f68a36731018d8c4b9cc871944.xml_TargetDevice=TEMPLATE_Config=() (13721 ms)
[----------] 1 test from conformance_other/ReadIRTest (13721 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (13721 ms total)
[  PASSED  ] 1 test.
TIME: 13.7786
```
- before
```
Note: Google Test filter = -*OpImplCheck*
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from conformance_other/ReadIRTest
[ RUN      ] conformance_other/ReadIRTest.ReadIR/PRC=_IR_name=Add_4560d07b463570c1fe7087406160cf403db2a3f68a36731018d8c4b9cc871944.xml_TargetDevice=TEMPLATE_Config=()

MEM_USAGE=70512KB
Access PluginCache ov core. OV Core use count: 0
Created ov core.
abs_threshold: 0.30426 rel_threshold: 1.79769e+308
[       OK ] conformance_other/ReadIRTest.ReadIR/PRC=_IR_name=Add_4560d07b463570c1fe7087406160cf403db2a3f68a36731018d8c4b9cc871944.xml_TargetDevice=TEMPLATE_Config=() (60014 ms)
[----------] 1 test from conformance_other/ReadIRTest (60014 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (60014 ms total)
[  PASSED  ] 1 test.
TIME: 60.0749
```

### Tickets:
 - *96799*
 - *93337*
 - *96595*

### CODE REVIEW

File: src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp [hunk 1]
```diff
@@ -322,8 +324,12 @@ std::vector<ov::Tensor> SubgraphBaseTest::get_plugin_outputs() {
 }
 
 void SubgraphBaseTest::validate() {
-    auto expectedOutputs = calculate_refs();
-    const auto& actualOutputs = get_plugin_outputs();
+    std::vector<ov::Tensor> expectedOutputs, actualOutputs;
+
+    std::thread t_device([&]{ actualOutputs = get_plugin_outputs(); });
```
Comment (EgorDuplensky) [inline src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:329]: Is it possible to have an option to disable async execution using
an environment variable for example?
I think it is highly likely that this async execution might make it harder to debug / troubleshoot issues.

Comment (iefode) [reply src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:333]: done

**Comment by EgorDuplensky** [comment]
@iefode I have also made a prototype for async execution of the tests.
In theory we could also make compile_model() step asynchronous.
Like:
1) start async ngraph ref inference and async compile_model()
2) as soon compile_model is finished -> start async plugin's infer() 
What do you think?

**Comment by iefode** [comment]
> @iefode I have also made a prototype for async execution of the tests. In theory we could also make compile_model() step asynchronous. Like:
> 
> 1. start async ngraph ref inference and async compile_model()
> 2. as soon compile_model is finished -> start async plugin's infer()
> 
> What do you think?

I think that is good idea. But the most performance increasing connected with `compare`. I think we can continue increasing test performance after this merge

**Comment by iefode** [comment]
@ilya-lavrenov Could you please merge the changes? ClDNN_UNIT_TESTS executable was failed, but my changes are not related to them. I want to merge the PR before nightly to check the changes impact :)

**Pull request merged**